### PR TITLE
resgroup: cleanup TODO/FIXME/XXX marks

### PR DESCRIPTION
**Pull request by pivotal-ning-yu**

### CODE REVIEW

File: src/backend/utils/resgroup/resgroup.c [hunk 1]
```diff
@@ -3225,6 +3236,46 @@ groupWaitQueueIsEmpty(const ResGroupData *group)
 	return waitQueue->size == 0;
 }
 
+#ifdef USE_ASSERT_CHECKING
```
Comment (xiong-gang) [inline src/backend/utils/resgroup/resgroup.c]: can we move all `USE_ASSERT_CHECKING` functions to the end of file and wrap them with one single `USE_ASSERT_CHECKING`

File: src/backend/utils/resgroup/resgroup.c [hunk 2]
```diff
@@ -3374,7 +3342,7 @@ resgroupDumpWaitQueue(StringInfo str, PROC_QUEUE *queue)
 	{
 		appendStringInfo(str, "{");
 		appendStringInfo(str, "\"pid\":%d,", proc->pid);
-		appendStringInfo(str, "\"resWaiting\":%d,", proc->resWaiting);
+		appendStringInfo(str, "\"resWaiting\":%d,", procIsWaiting(proc));
```
Comment (kainwen) [inline src/backend/utils/resgroup/resgroup.c]: To be consistent with previous bool dump, I think that it is better to turn it to "true" or "false" string.

Comment (kainwen) [reply src/backend/utils/resgroup/resgroup.c]: Others LGTM.

File: src/backend/utils/resgroup/resgroup.c [hunk 3]
```diff
@@ -1638,112 +1640,27 @@ groupAcquireSlot(ResGroupData *group)
...
- *       new memQuotaGranted is 0.264 - 0.016 = 0.248
- *       new memStocksFree is 0.144 - 0.016 = 0.128
- *   for memShared we could free currentSharedUsage - Max(currentSharedUsage, 0.4*0.6) = -0.18
- *
- * * memLimit: 0.4 -> 0.2
- *   for memQuotaGranted we could free 0.128 - 0.2*0.4/20*16 = 0.128 - 0.064 = 0.064
- *       new memQuotaGranted is 0.248-0.064 = 0.184
- *       new memStocksFree is 0.128 - 0.064 = 0.064
- *   for memShared we could free currentSharedStocks - Max(currentSharedUsage, 0.2*0.6) = -0.04
- */
```
[the remark this answers is not in the document]
Comment (pivotal-ning-yu) [reply src/backend/utils/resgroup/resgroup.c]: sure, let me add it back.

File: src/backend/utils/resgroup/resgroup.c (hunk 2 above)
Comment (pivotal-ning-yu) [reply src/backend/utils/resgroup/resgroup.c]: makes sense.

File: src/backend/utils/resgroup/resgroup.c (hunk 1 above)
Comment (pivotal-ning-yu) [reply src/backend/utils/resgroup/resgroup.c]: In fact these `USE_ASSERT_CHECKING` functions was wrapped separately intentionally, the consideration was that:

- these function are not designed to be debug only, some of them might not need the `USE_ASSERT_CHECKING` flag some day, so if we move all of them together now then shall we move them out at that time? And if a function became debug only in the future shall we move them in?
- currently most functions are grouped by class (self, group, groupWaitQueue, etc.), which makes more sense from my point of view, in fact I also suggest to split resgroup.c one day into resgroup-slot.c, resgroup-wait-queue.c etc. in the future as resgroup.c is too long now. If we really decided to do so, shall we have a resgroup-debug-only.c or so for these debug functions?

I have similar comments in another PR in which you move these functions together.

Anyway as that PR was merged with those comments ignored, I think I have to also do the same here to keep consistency, but I still don't suggest to do so.

**Comment by pivotal-ning-yu** [comment]
rebased with origin/master to resolve the conflicts and updated according to comments.

File: src/backend/utils/resgroup/resgroup.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (pivotal-ning-yu) [reply src/backend/utils/resgroup/resgroup.c]: a commit is pushed to add them back. currently I keep them as the comments, although moved from `groupApplyMemCaps()` to `mempoolAutoRelease()`. if one day we want to move them to README or so we should put more details on the examples to explain the policy.

File: src/backend/utils/resgroup/resgroup.c [hunk 4]
```diff
@@ -3251,6 +3179,46 @@ groupWaitQueueIsEmpty(const ResGroupData *group)
...
+	groupWaitQueueValidate(group);
+
+	waitQueue = &group->waitProcs;
+	head = &waitQueue->links;
+	target = &proc->links;
+
+	for (iter = (SHM_QUEUE *) MAKE_PTR(head->next);
+		 iter != head;
+		 iter = (SHM_QUEUE *) MAKE_PTR(iter->next))
+	{
```
[the remark this answers is not in the document]
Comment (asimrp) [reply src/backend/utils/resgroup/resgroup.c]: Staying consistent with rest of the code and using `SHMQueueNext()` is better, I think.  An elaborate comment could be used to describe how it's working.

[the remark this answers is not in the document]
Comment (pivotal-ning-yu) [reply src/backend/utils/resgroup/resgroup.c]: Sure, thanks for the suggestion, I will change it later.

**Comment by pivotal-ning-yu** [comment]
rebased again to resolve conflicts.

**Pull request merged**